test_integration/03_redeem_and_unstake/utils/progress_revert_redeem_assertion.js

```diff
@@ -0,0 +1,242 @@
+
+    assert.strictEqual(
+      expectedRedeemerBaseTokenBalance.eq(finalBalances.baseToken.redeemer),
+      true,
+      `Redeemer base token balance must be ${expectedRedeemerBaseTokenBalance.toString(10)}`
+           + ` instead of ${finalBalances.baseToken.redeemer.toString(10)}`,
+    );
+
+    const expectedRedeemerTokenBalance = initialBalances.token.redeemer.add(redeemRequest.amount);
+
+    assert.strictEqual(
+      expectedRedeemerTokenBalance.eq(finalBalances.token.redeemer),
+      true,
+      `Redeemer token balance must be ${expectedRedeemerTokenBalance.toString(10)}`
+           + ` instead of ${finalBalances.token.redeemer.toString(10)}`,
+    );
+
+    // Assert burner balance
+    const expectedBurnerBaseTokenBalance = initialBalances.baseToken
+      .burner.add(redeemRequest.bounty);
+
+    assert.strictEqual(
+      expectedBurnerBaseTokenBalance.eq(finalBalances.baseToken.burner),
+      true,
+      `Burner base token balance must be ${
+        expectedBurnerBaseTokenBalance.toString(10)
+      } instead of ${finalBalances.baseToken.burner.toString(10)}`,
+    );
+
+    const expectedBurnerTokenBalance = initialBalances.token.burner;
+
+    assert.strictEqual(
+      expectedBurnerTokenBalance.eq(finalBalances.token.burner),
+      true,
+      `Burner token balance must be ${
+        expectedBurnerTokenBalance.toString(10)
+      } instead of ${finalBalances.token.burner.toString()}`,
+    );
+  }
+
+  /**
+   * This assert event after progress revert redeem method.
+   * @param {Object} event Event object after decoding.
+   * @param {RedeemRequest} redeemRequest Redeem request parameters.
+   * @private
+   */
+  static _assertProgressRevertRedeemEvent(event, redeemRequest) {
+    const eventData = event.RedeemReverted;
+
+    assert.strictEqual(
+      eventData._messageHash,
+      redeemRequest.messageHash,
+      `Message hash from the event must be equal to ${
+        redeemRequest.messageHash
+      }.`,
+    );
+
+    assert.strictEqual(
+      eventData._redeemer,
+      redeemRequest.redeemer,
+      `Redeemer address from the event must be equal to ${
+        redeemRequest.redeemer
+      }.`,
+    );
+
+    assert.strictEqual(
+      eventData._redeemerNonce.eq(redeemRequest.nonce),
+      true,
+      `Redeemer nonce ${eventData._redeemerNonce.toString(
+        10,
+      )} from the event must be equal to ${redeemRequest.nonce.toString(
+        10,
+      )}.`,
+    );
+
+    assert.strictEqual(
+      eventData._amount.eq(redeemRequest.amount),
+      true,
+      `Redeem amount ${eventData._amount.toString(
+        10,
+      )} from the event must be equal to ${redeemRequest.amount.toString(
+        10,
+      )}.`,
+    );
+  }
+
+  /**
+   * Returns ETH balance wrapped in BN.
+   * @param {string} address Address for which balance is requested.
+   * @return {Promise<BN>} ETH Balance.
+   * @private
+   */
+  async _getEthBalance(address) {
```

Comment:
I think there is a risk of confusion referring to both base token and eth. In the absence of a substantive reason that outweighs that risk of confusion, I suggest renaming this to something like `_getBaseTokenBalance` or some other consistency solution.

🤓 I acknowledge that this function is defined in a few places and that the best solution may be to leave it as is here and instead make this (and other deduplication refactoring changes) in one or more other PRs.

**I'm also curious about the value of this function.** Is the below too clunky or bad practice:
```
new BN(await this.web3.eth.getBalance(<address>));
```